GDJS/Runtime/CustomRuntimeObject.ts

```diff
@@ -0,0 +1,587 @@
+      this.doStepPostEvents(instanceContainer);
+      if (profiler) {
+        profiler.end(this._objectData.type);
+      }
+
+      this._instanceContainer._updateObjectsPostEvents();
+    }
+
+    /**
+     * This method is called when the preview is being hot-reloaded.
+     */
+    onHotReloading(instanceContainer: gdjs.RuntimeInstanceContainer) {}
+
+    // This is only to handle trigger once.
+    doStepPreEvents(instanceContainer: gdjs.RuntimeInstanceContainer) {}
+
+    /**
+     * This method is called each tick after events are done.
+     * @param instanceContainer The instanceContainer owning the object
+     */
+    doStepPostEvents(instanceContainer: gdjs.RuntimeInstanceContainer) {}
+
+    /**
+     * This method is called when the object is being removed from its parent
+     * container and is about to be destroyed/reused later.
+     */
+    onDestroy(instanceContainer: gdjs.RuntimeInstanceContainer) {}
+
+    updatePreRender(instanceContainer: gdjs.RuntimeInstanceContainer): void {
+      this._instanceContainer._updateObjectsPreRender();
+      this.getRenderer().ensureUpToDate();
+    }
+
+    getRendererObject() {
+      return this.getRenderer().getRendererObject();
+    }
+
+    getRenderer() {
+      return this._instanceContainer.getRenderer();
+    }
+
+    onChildrenLocationChanged() {
+      this._isUntransformedHitBoxesDirty = true;
+      this.invalidateHitboxes();
+      this.getRenderer().update();
+    }
+
+    updateHitBoxes(): void {
+      if (this._isUntransformedHitBoxesDirty) {
+        this._updateUntransformedHitBoxes();
+      }
+
+      //Update the current hitboxes with the frame custom hit boxes
+      //and apply transformations.
+      for (let i = 0; i < this._untransformedHitBoxes.length; ++i) {
+        if (i >= this.hitBoxes.length) {
+          this.hitBoxes.push(new gdjs.Polygon());
+        }
+        for (
+          let j = 0;
+          j < this._untransformedHitBoxes[i].vertices.length;
+          ++j
+        ) {
+          if (j >= this.hitBoxes[i].vertices.length) {
+            this.hitBoxes[i].vertices.push([0, 0]);
+          }
+          this.applyObjectTransformation(
+            this._untransformedHitBoxes[i].vertices[j][0],
+            this._untransformedHitBoxes[i].vertices[j][1],
+            this.hitBoxes[i].vertices[j]
+          );
+        }
+        this.hitBoxes[i].vertices.length = this._untransformedHitBoxes[
+          i
+        ].vertices.length;
+      }
+    }
+
+    /**
+     * Merge the hitboxes of the children.
+     */
+    _updateUntransformedHitBoxes() {
+      this._isUntransformedHitBoxesDirty = false;
+
+      const oldUnscaledCenterX =
+        (this._unrotatedAABB.max[0] + this._unrotatedAABB.min[0]) / 2;
+      const oldUnscaledCenterY =
+        (this._unrotatedAABB.max[1] + this._unrotatedAABB.min[1]) / 2;
+
+      this._untransformedHitBoxes.length = 0;
+      if (this._instanceContainer.getAdhocListOfAllInstances().length === 0) {
```

Comment:
We should probably avoid calling getAdhocListOfAllInstances() twice, as it's not a trivial operation.

Reply:
Ok the result is actually memoized so no need!

Reply:
It is now, it only rebuilds it if necessary. But you're right, I'll do the change.